Helper/PaymentMethods.php

```diff
@@ -155,27 +165,39 @@ public function getPaymentMethods($quoteId, $country = null)
         return $this->fetchPaymentMethods($country);
     }
 
-    /**
-     * @return array
-     */
-    public function getAdyenPaymentCodes(): array
-    {
-        return [self::ADYEN_HPP, self::ADYEN_CC, self::ADYEN_ONE_CLICK];
-    }
-
     /**
      * @param string $methodCode
      * @return bool
      */
     public function isAdyenPayment(string $methodCode): bool
     {
-        if(in_array($methodCode, $this->getAdyenPaymentCodes())) {
+        if(in_array($methodCode, $this->getAdyenPaymentMethods())) {
             return false;
         }
 
         return true;
     }
 
+    /**
+     * Returns an array of Adyen payment method codes
+     *
+     * @return string[]
+     */
+    public function getAdyenPaymentMethods() : array
+    {
+        $paymentMethods = $this->dataHelper->getPaymentMethodList();
+
+        $filtered = array_filter(
+            $paymentMethods,
+            function ($key) {
+                return str_starts_with($key, self::ADYEN_PREFIX);
```

Comment:
The `str_starts_with` function is from PHP 8. Since we support Magento 2.3 and PHP 7, this might need some tweaks.

Reply:
Thanks for the comment @michaelpaul. I will use `strpos()` for this one.